src/format/opamFile.ml

```diff
@@ -3253,6 +3253,13 @@ module OPAM = struct
 
   let effective_part ?(modulo_state=false) (t:t) =
     let t_modulo_state = if modulo_state then empty else t in
+    let effective_url u =
+      match URL.checksum u with
+      | [] -> URL.create (URL.url u) (* ignore mirrors *)
```

Comment:
We could try to take the smallest one. I feel like there is more of a use-case for avoiding rebuilds if we want to add better checksums than the opposite. e.g. if someday we modify all the opam packages to have a mandatory `sha512` it would be nice to avoid this change making everyone rebuild everything

Reply:
good for me

Reply:
Done in https://github.com/ocaml/opam/pull/5258/commits/2a0465b5f3702c3f4b7afdc8df87da3739dc1b32

Reply:
actually i reverted it. It feels like taking the first one in the list is a good default as people usually add new checksums at the end of the list. There is a case to be done about adding lower strength checksums too (e.g. to avoid the openssl dependencies when using opam < 2.2).

Reply:
make sense